src/Model.d.ts

```diff
@@ -101,9 +103,14 @@ type EntityField<T extends OneTypedField> =
 /*
     Entities are typed objects whoes signature is based on the schema model of the same name.
  */
-export type Entity<T extends OneTypedModel> = {
-    [P in keyof T]?: EntityField<T[P]>
-};
+export type Entity<T extends OneTypedModel> = UndefinedToOptional<{
+    [P in keyof T]: T[P]['required'] extends true ? EntityField<T[P]> : EntityField<T[P]> | undefined
```

Comment:
Can you please add a comment to the file and explain here how this works.

Reply:
DITO.  I wouldnt have been able to create the UndefinedToOptional but i can definitely read it and explain it.

this new extends thing is also new to me, but its basically what you do in the following code, basically if a hardcoded value is of a certain type then the corresponding type can be derived .  I think eventually some more typing should be added for values that have a default value.

```
type EntityField<T extends OneTypedField> =
      T['type'] extends ArrayConstructor ? any[]
    : T['type'] extends BooleanConstructor ? boolean
    : T['type'] extends NumberConstructor ? number
    : T['type'] extends ObjectConstructor ? object
    : T['type'] extends DateConstructor ? Date
    : T['type'] extends StringConstructor ? string
    : T['type'] extends SetConstructor ? Set<T>
    : never;
```

I added comments. to the code

Reply:
The problem for default is that it can be a function. i.e. determined at runtime.

I'm having trouble with the line:

    [P in keyof T]: T[P]['required'] extends true ? EntityField<T[P]> : EntityField<T[P]> | undefined

Is that saying .... If there is a property required that is truthy (extends true), then the result is EntityField<...> otherwise it can be EntityField or undefined.?

Reply:
Yes. Although im not sure if it's truthy or true. In this case it can't be truthy because the field required can only be a boolean